projects/step-frontend/src/lib/modules/execution/components/alt-execution-progress/alt-execution-progress.component.ts

```diff
@@ -174,44 +158,23 @@ export class AltExecutionProgressComponent implements OnInit, OnDestroy, AltExec
 
   protected readonly _executionMessages = inject(ViewRegistryService).getDashlets('execution/messages');
 
-  private isTreeInitialized = false;
-
-  private relativeTime = signal<number | undefined>(undefined);
+  private timeRangeSelection$: BehaviorSubject<TimeRangePickerSelection> =
+    new BehaviorSubject<TimeRangePickerSelection>(undefined!);
 
-  updateRelativeTime(time?: number) {
-    this.relativeTime.set(time);
-  }
+  readonly timeRangeChange$ = this.timeRangeSelection$.asObservable().pipe(skip(1), shareReplay(1));
```

Comment:
@lucianbaciu 
`shareReplay` adds another subject under the hood. Please add `takeUntilDestroyed` to the pipe